### ISSUE TITLE
Cannot tab to elements with [role=button] with default MacOS and Firefox

### ISSUE THREAD

**Issue by NickColley** [alphagov/reported-bugs#43]
**Upstream bug:** https://bugzilla.mozilla.org/show_bug.cgi?id=1560117
**Date:** 2019-07-19
**Reported by:** Nick Colley
**Related to:** https://github.com/alphagov/govuk-frontend/issues/1464

---

### Overview

In some circumstances website authors will use aria roles to indicate that a link that looks like a button should be announced as a button.

This is useful so that assistive technologies such as dictation software can access this button, for example "click button".

In Firefox on MacOS these buttons are not part of the tab order by default, which could cause barriers.

#### Steps to reproduce
1. Use MacOS (version 10.13.6 (17G7024))
2. Ensure 'full keyboard access' is on the default "text boxes and lists only"
You can find this setting by going to 'Systems Preferences' > 'Keyboard' > 'Shortcuts'.
3. Open example https://output.jsbin.com/lozamem
4. Tab into the example, between items

### Expected result

Both the button with the <button> markup and the 'link button' with [role=button] should be tabbable

### Actual result

Only the button with <button> markup is tabbable

**Comment by NickColley** [comment]
Marco:

"Yes, this is WONTFIX. Unless the ARIA spec some day calls for browsers to change their behavior at the presence of role="button" or role="link", or any actionable widget, for that matter, browsers will not automatically tab-stop on such elements unless tabindex="0" is set as well, and the operating system, like MacOS before Catalina, is set to stop at all focusable elements, not just text fields. Catalina seems to no longer make that distinction, AKA it will stop at all focusable elements always, like Windows and Linux do. Closing as WONTFIX."
